File: day01p2.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, err := os.Open("day01.input")
	if err != nil {
		panic(err)
	}

	floor := 0
	position := 1
	b1 := make([]byte, 1)
	for {
		_, err := f.Read(b1)
		if err != nil {
			break
		}

		if string(b1) == "(" {
			floor++
		} else if string(b1) == ")" {
			floor--
			if floor == -1 {
				fmt.Println(position)
			}
		}

		position++
	}
}

File: perl5/day01p2.pl
#!/usr/bin/perl
use strict;
use warnings;

open my $file, "<", "../day01.input"
	or die $!;

my $floor = 0;
my $position = 1;
while(my $line = <$file>) {
	chomp $line;
	for my $char (split //, $line) {
		$floor++ if $char eq "(";
		$floor-- if $char eq ")";
		if ($floor == -1) {
			print "$position\n"; 
		}

		$position++;
	}
}
